src/EthereumController.sol

```diff
@@ -16,6 +16,7 @@ interface IDaiNstLike {
 interface ISNSTLike {
```

Comment:
Instead of manually adding ERC4626 functions, I would just inherit the forge-std `IERC4626`. In fact `deposit` is not even correct - it should return shares. Also, the `IERC20` can be changed to:

`import { IERC20 } from "forge-std/interfaces/IERC20.sol";`

`erc20-helpers` can actually be removed as a dep because we only need the interface which forge-std provides.

Reply:
Updated and removed